spring-batch-2/src/main/java/com/baeldung/batch/BatchConfiguration.java

```diff
@@ -4,31 +4,24 @@
 
 import org.springframework.batch.core.Job;
 import org.springframework.batch.core.Step;
-import org.springframework.batch.core.configuration.annotation.EnableBatchProcessing;
-import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
-import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.core.job.builder.JobBuilder;
 import org.springframework.batch.core.launch.support.RunIdIncrementer;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.core.step.builder.StepBuilder;
 import org.springframework.batch.item.database.BeanPropertyItemSqlParameterSourceProvider;
 import org.springframework.batch.item.database.JdbcBatchItemWriter;
 import org.springframework.batch.item.database.builder.JdbcBatchItemWriterBuilder;
 import org.springframework.batch.item.file.FlatFileItemReader;
 import org.springframework.batch.item.file.builder.FlatFileItemReaderBuilder;
 import org.springframework.batch.item.file.mapping.BeanWrapperFieldSetMapper;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.io.ClassPathResource;
+import org.springframework.transaction.PlatformTransactionManager;
 
 @Configuration
```

Comment:
Is there any reason to remove @EnableBatchProcessing, that is still suggested in v5, and helps create the JobRepository?
The only reason I see to remove this maybe is that in SpringBootBatchIntegrationTest we use @EnableAutoConfiguration that configures batch-related stuff under the hood.
But, for clarity and performance reasons, lets use:

@Configuration
@EnableBatchProcessing
public class BatchConfiguration {

    @Bean(name = "dataSource")
    public DataSource getDataSource() {
        DataSourceBuilder dataSourceBuilder = DataSourceBuilder.create();
        dataSourceBuilder.driverClassName("org.h2.Driver");
        dataSourceBuilder.url("jdbc:h2:mem:test");
        return dataSourceBuilder.build();
    }

    @Bean(name = "transactionManager")
    public PlatformTransactionManager getTransactionManager() {
        return new ResourcelessTransactionManager();
    }
...


and in SpringBootBatchIntegrationTest:

@SpringJUnitConfig(BatchConfiguration.class)
@SpringBatchTest
@TestPropertySource("classpath:application.properties")
@DirtiesContext
@Sql({ "classpath:org/springframework/batch/core/schema-h2.sql", "classpath:schema-all.sql" })
public class SpringBootBatchIntegrationTest {
...

Also create a new application.properties in test/resources (duplicate of main/resources)

Reply:
Based on this migration guide for Spring Boot 3.0 @EnableBatchProcessing is now discouraged.